lib/stdlib/src/proc_lib.erl

```diff
@@ -668,65 +683,108 @@ clean_dict([{'$ancestors',_}|Dict]) ->
     clean_dict(Dict);
 clean_dict([{'$initial_call',_}|Dict]) ->
     clean_dict(Dict);
+clean_dict([{'$process_description',_}|Dict]) ->
+    clean_dict(Dict);
 clean_dict([E|Dict]) ->
     [E|clean_dict(Dict)];
 clean_dict([]) ->
     [].
 
 get_dictionary(Pid,Tag) ->
-    case get_process_info(Pid,dictionary) of
-	{dictionary,Dict} ->
-	    case lists:keysearch(Tag,1,Dict) of
-		{value,Value} -> Value;
-		_             -> undefined
-	    end;
+    try get_process_info(Pid, {dictionary, Tag}) of
+	{{dictionary,Tag},Value} ->
+            Value;
 	_ ->
 	    undefined
+    catch _:_ -> %% rpc to old node
+            case get_process_info(Pid,dictionary) of
+                {dictionary,Dict} ->
+                    case lists:keysearch(Tag,1,Dict) of
+                        {value,Value} -> Value;
+                        _ -> undefined
+                    end;
+                _ ->
+                    undefined
+            end
     end.
 
 linked_info(Pid) ->
   make_neighbour_reports1(neighbours(Pid)).
   
 make_neighbour_reports1([P|Ps]) ->
-  ReportBody = make_neighbour_report(P),
-  %%
-  %%  Process P might have been deleted.
-  %%
-  case lists:member(undefined, ReportBody) of
-    true ->
-      make_neighbour_reports1(Ps);
-    false ->
-      [{neighbour, ReportBody}|make_neighbour_reports1(Ps)]
-  end;
+    %%
+    %%  Process P might have been deleted.
+    %%
+    case make_neighbour_report(P) of
+        undefined ->
+            make_neighbour_reports1(Ps);
+        ReportBody ->
+            [{neighbour, ReportBody}|make_neighbour_reports1(Ps)]
+    end;
 make_neighbour_reports1([]) ->
-  [].
+    [].
   
 %% Do not include messages or process dictionary, even if
 %% error_logger_format_depth is unlimited.
 make_neighbour_report(Pid) ->
-  [{pid, Pid},
-   get_process_info(Pid, registered_name),          
-   get_initial_call(Pid),
-   get_process_info(Pid, current_function),
-   get_ancestors(Pid),
-   get_process_info(Pid, message_queue_len),
-   %% get_messages(Pid),
-   get_process_info(Pid, links),
-   %% get_cleaned_dictionary(Pid),
-   get_process_info(Pid, trap_exit),
-   get_process_info(Pid, status),
-   get_process_info(Pid, heap_size),
-   get_process_info(Pid, stack_size),
-   get_process_info(Pid, reductions),
-   get_process_info(Pid, current_stacktrace)
-  ].
- 
-get_initial_call(Pid) ->
-    case get_dictionary(Pid, '$initial_call') of
-	{'$initial_call', {M, F, A}} ->
+    Keys = [registered_name,
+            initial_call, current_function,
+            message_queue_len, links, trap_exit,
+            status, heap_size, stack_size, reductions,
+            current_stacktrace
+           ],
+    PInfo = get_process_info(Pid, Keys),
+    
+    DictKeys = [{dictionary, '$process_description'},
+                {dictionary, '$initial_call'},
+                {dictionary, '$ancestors'}],
+
+    DInfo = try get_process_info(Pid, DictKeys)
```

Comment:
why not `DictInfo` to be little more consistent in naming or `DKeys`?
I'm a bit picky, but I think consistent code is easier to read - I hope I'm so consistent myself :-).

Please ignore if you think I'm too picky ;-)
I like 1 char variables from time to time and when they are obvious. What I dislike here is 2 mixed conventions.

Reply:
That is picky :-)

Reply:
Fixed